packages/kbn-esql-ast/src/parser/README.md

```diff
@@ -1,6 +1,91 @@
+# ES|QL Parser
+
+The Kibana ES|QL parser uses the ANTLR library for lexing and parse tree (CST)
+generation. The ANTLR grammar is imported from the Elasticsearch repository in
+an automated CI job.
+
+We use the ANTLR outputs: (1) the token stream; and (2) the parse tree to
+generate (1) the Abstract Syntax Tree (AST), (2) for syntax validation, (3) for
+syntax highlighting, and (4) for formatting (comment and whitespace) extraction
+and assignment to AST nodes.
+
+In general ANTLR is resilient to grammar errors, in the sense that it can
+produce a Parser tree up to the point of the error, then stops. This is useful
+to perform partial tasks even with broken queries and this means that a partial
+AST can be produced even with an invalid query.
+
+
+## Folder structure
+
+The parser is structured as follows:
+
+```
+src/
+|- parser/                            Contains the logic to parse the ES|QL query and generate the AST.
+|  |- factories.ts                    Contains AST node factories.
+|  |- antlr_error_listener.ts         Contains code which traverses ANTLR CST and collects syntax errors.
+|  |- esql_ast_builder_listener.ts    Contains code which traverses ANTLR CST and builds the AST.
+|
+|- antlr/                             Contains the autogenerated ES|QL ANTLR grammar files and various compilation assets.
+   |- esql_lexer.g4                   Contains the ES|QL ANTLR lexer grammar.
+   |- esql_parser.g4                  Contains the ES|QL ANTLR parser grammar.
+```
+
+
+## Usage
+
+### Get AST from a query string
+
+The `parse` function returns the AST data structure, unless a syntax error
+happens in which case the `errors` array gets populated with a Syntax errors.
+
+```js
+import { parse } from '@kbn/esql-ast';
+
+const queryString = "FROM index | STATS 1 + AVG(myColumn) ";
+const { root, errors } = await astProvider(queryString);
```

Comment:
Shouldn't here be parse instead of astProvider? (at least this is imported)

Reply:
Yes, thank you, copy-paste error! :)